storage/innobase/include/arch0arch.h

```diff
@@ -0,0 +1,1984 @@
+  is one file context for a archive group.
+  @param[in]  path        path to the file
+  @param[in]  base_dir    directory name prefix
+  @param[in]  base_file   file name prefix
+  @param[in]  num_files   initial number of files
+  @param[in]  file_size   size of file used when a new file is created
+  @param[in]  uuid        uuid of this arch group or 0 if unknown
+  @return error code. */
+  dberr_t init_file_ctx(const char *path, const char *base_dir,
+                        const char *base_file, uint num_files,
+                        uint64_t file_size, Arch_group_uuid uuid)
+  {
+    m_uuid= uuid;
+    m_file_size= file_size;
+    return (m_file_ctx.init(path, base_dir, base_file, num_files));
+  }
+
+  /* Close the file contexts when they're not required anymore. */
+  void close_file_ctxs()
+  {
+    m_file_ctx.close();
+
+    if (m_durable_file.m_file != OS_FILE_CLOSED)
+    {
+      os_file_close(m_durable_file);
+      m_durable_file.m_file = OS_FILE_CLOSED;
+    }
+  }
+
+  /** Mark archive group inactive.
+  A group is marked inactive by archiver background before entering
+  into idle state ARCH_STATE_IDLE.
+  @param[in]    end_lsn lsn where redo archiving is stopped */
+  void disable(lsn_t end_lsn) {
+    m_is_active= false;
+
+    if (end_lsn != LSN_MAX)
+      m_end_lsn= end_lsn;
+  }
+
+  /** Attach a client to the archive group.
+  @param[in]    is_durable      true, if durable tracking is requested */
+  void attach(bool is_durable)
+  {
+    mysql_mutex_assert_owner(m_arch_mutex);
+    ++m_num_active;
+
+    if (is_durable)
+      ++m_dur_ref_count;
+    else
+    {
+      ut_ad(m_ref_count < std::numeric_limits<decltype(m_ref_count)>::max());
+      if (m_ref_count < std::numeric_limits<decltype(m_ref_count)>::max())
+        ++m_ref_count;
+    }
+  }
+
+  /** Detach a client when archiving is stopped by the client.
+  The client still has reference to the group so that the group
+  is not destroyed when it retrieves the archived data. The
+  reference is removed later by #Arch_Group::release.
+  @param[in]    stop_lsn        archive stop lsn for client
+  @param[in]    stop_pos        archive stop position for client. Used only by
+  the page_archiver.
+  @return number of active clients */
+  uint detach(lsn_t stop_lsn, Arch_Page_Pos *stop_pos)
+  {
+    ut_ad(m_num_active > 0);
+    mysql_mutex_assert_owner(m_arch_mutex);
+    --m_num_active;
+
+    if (m_num_active == 0)
+    {
+      m_end_lsn = stop_lsn;
+      if (stop_pos != nullptr)
+        m_stop_pos = *stop_pos;
+    }
+    return m_num_active;
+  }
+
+  /** Release the archive group from a client.
+  Reduce the reference count. When all clients release the group,
+  the reference count falls down to zero. The function would then
+  return zero and the caller can remove the group.
+  @param[in]    is_durable      the client needs durable archiving */
+  void release(bool is_durable)
+  {
+    mysql_mutex_assert_owner(m_arch_mutex);
```

Comment:
This is unused at least in some expansions of the inline function. Do we ever have `is_durable=true`?